documentation/proposals/Proposal - Generic Math.md

```diff
@@ -0,0 +1,1182 @@
+
+    /// <summary>Returns the quaternion that results from multiplying two quaternions together.</summary>
+    /// <param name="value1">The first quaternion.</param>
+    /// <param name="value2">The second quaternion.</param>
+    /// <returns>The product quaternion.</returns>
+    /// <remarks>The <see cref="Quaternion.op_Multiply" /> method defines the operation of the multiplication operator for <see cref="Quaternion" /> objects.</remarks>
+    public static Quaternion<T> operator *(Quaternion<T> value1, Quaternion<T> value2);
+
+    /// <summary>Returns the quaternion that results from scaling all the components of a specified quaternion by a scalar factor.</summary>
+    /// <param name="value1">The source quaternion.</param>
+    /// <param name="value2">The scalar value.</param>
+    /// <returns>The scaled quaternion.</returns>
+    /// <remarks>The <see cref="Quaternion.op_Multiply" /> method defines the operation of the multiplication operator for <see cref="Quaternion" /> objects.</remarks>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static Quaternion<T> operator *(Quaternion<T> value1, T value2);
+
+    /// <summary>Subtracts each element in a second quaternion from its corresponding element in a first quaternion.</summary>
+    /// <param name="value1">The first quaternion.</param>
+    /// <param name="value2">The second quaternion.</param>
+    /// <returns>The quaternion containing the values that result from subtracting each element in <paramref name="value2" /> from its corresponding element in <paramref name="value1" />.</returns>
+    /// <remarks>The <see cref="op_Subtraction" /> method defines the operation of the subtraction operator for <see cref="Quaternion" /> objects.</remarks>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static Quaternion<T> operator -(Quaternion<T> value1, Quaternion<T> value2);
+
+    /// <summary>Reverses the sign of each component of the quaternion.</summary>
+    /// <param name="value">The quaternion to negate.</param>
+    /// <returns>The negated quaternion.</returns>
+    /// <remarks>The <see cref="op_UnaryNegation" /> method defines the operation of the unary negation operator for <see cref="Quaternion" /> objects.</remarks>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static Quaternion<T> operator -(Quaternion<T> value);
+
+    /// <summary>Adds each element in one quaternion with its corresponding element in a second quaternion.</summary>
+    /// <param name="value1">The first quaternion.</param>
+    /// <param name="value2">The second quaternion.</param>
+    /// <returns>The quaternion that contains the summed values of <paramref name="value1" /> and <paramref name="value2" />.</returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static Quaternion<T> Add<T>(Quaternion<T> value1, Quaternion<T> value2)
+        where T : IRootFunctions<T>, ITrigonometricFunctions<T>;
```

Comment:
It makes no sense for this to be a generic method in a generic type.

Reply:
Does this even need to exist given `operator +` exists?